orttraining/orttraining/python/training/ortmodule/_utils.py

```diff
@@ -38,21 +38,24 @@ def _ortvalue_from_torch_tensor(torch_tensor):
     return C.OrtValue.from_dlpack(to_dlpack(torch_tensor), is_bool_tensor)
 
 
-def _torch_tensor_from_dl_pack(dlpack, ortvalue, device):
-    torch_tensor = from_dlpack(dlpack) if device.type != 'ort' else C.ort_from_dlpack(dlpack)
+def _torch_tensor_from_dl_pack(dlpack, ortvalue):
+    torch_tensor = from_dlpack(dlpack)
     return torch_tensor.to(torch.bool) if ortvalue.data_type() == 'tensor(bool)' else torch_tensor
 
 
 def _ortvalue_to_torch_tensor(ortvalue, device):
+    if device.type == 'ort':
+        return C.to_aten_ort_device_tensor(ortvalue)
     # PyTorch's to_dlpack() uses same config for both torch.bool and torch.uint8,
     # and convert the config to torch.uint8 tensor duing from_dlpack().
     # So we need to convert the torch tensor to torch.bool type if OrtValue is bool tensor.
     dlpack_tensor = ortvalue.to_dlpack()
-    return _torch_tensor_from_dl_pack(dlpack_tensor, ortvalue, device)
+    return _torch_tensor_from_dl_pack(dlpack_tensor, ortvalue)
 
-def _torch_tensor_to_dlpack(tensor):
+# convert torch tensor to ort accept tensors, could be dlpack or ortvalue
+def _torch_tensor_to_ort_accept_tensors(tensor):
```

Comment:
Having a function that either returns a `ORTValue` or a `DLPack` tensor is confusing.
Ideally it should return the same type in both branches, so callers know exactly what to expect

Reply:
I can move the branches to the caller to eliminate the confusing, but that will make the caller looks more complicated.

Reply:
Overall changes are OK, my only concern is the multi-return aspect of `_torch_tensor_to_ort_accept_tensors`

Having conditional return types probably kill JIT's compiler optimization and in general as a bad idea for the reasons many languages use static typing, such as C++. Having to refactor an API to fit this duality really increases coupling and makes maintenance hard down the line.

I'd consider simplifying implementation to use specific functions for specific tasks and use external logic/pattern to remove ambiguity

Reply:
update the implementation, to make it a explicit condition in the caller.